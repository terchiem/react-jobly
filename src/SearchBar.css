.SearchBar {
  display: flex;
  justify-content: center;
}

.SearchBar-input {
  width: 100%;
  padding: 8px 10px;
}

.SearchBar-button {
  padding: 10px 20px;
}