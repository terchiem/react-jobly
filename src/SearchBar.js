import React, { useState } from 'react';
import './SearchBar.css';

/** Search bar component to search for a Company or Job */

function SearchBar({ search }) {  

  const [searchTerm, setSearchTerm] = useState('');

  function handleSubmit(evt) {
    evt.preventDefault();
    search(searchTerm);
    setSearchTerm('');
  }

  // TODO: debounce the search
  function handleChange(evt) {
    setSearchTerm(evt.target.value);
  }

  return (
    <form className="SearchBar" onSubmit={handleSubmit}>
      <input 
        className="SearchBar-input" 
        placeholder="Search term..."
        value={searchTerm}
        onChange={handleChange}
      />
      <button className="SearchBar-button">Search</button>
    </form>
  )
}

export default SearchBar;