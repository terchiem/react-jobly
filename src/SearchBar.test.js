import React from 'react';
import { render } from '@testing-library/react';
import SearchBar from './SearchBar';

test('renders without crashing', () => {
  render(<SearchBar />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<SearchBar />);
  expect(asFragment()).toMatchSnapshot();
});