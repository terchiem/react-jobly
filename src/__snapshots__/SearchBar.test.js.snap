// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="SearchBar"
  >
    <input
      class="SearchBar-input"
      placeholder="Search term..."
      value=""
    />
  </form>
</DocumentFragment>
`;
